hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/txns/contract/ContractCreateTransitionLogicTest.java

```diff
@@ -1081,6 +1095,70 @@ void followsHappyPathWithCounterAndRecordAndSidecars() {
         assertEquals(10, (int) changes.get(MAX_AUTOMATIC_ASSOCIATIONS));
     }
 
+    @Test
+    void followsHappyPathForContractNoncesWithOverrides() {
+        // setup:
+        givenValidTxnCtxWithMaxAssociations();
+        final var captor = ArgumentCaptor.forClass(ContractCustomizer.class);
+        final var secondaryCreations = List.of(IdUtils.asContract("0.0.849321"));
+        given(properties.isContractsNoncesExternalizationEnabled()).willReturn(true);
+        // and:
+        given(accountStore.loadAccount(senderAccount.getId())).willReturn(senderAccount);
+        given(hfs.cat(bytecodeSrc)).willReturn(bytecode);
+        given(accessor.getTxn()).willReturn(contractCreateTxn);
+        given(txnCtx.activePayer()).willReturn(ourAccount());
+        given(txnCtx.accessor()).willReturn(accessor);
+        given(worldState.getCreatedContractIds()).willReturn(secondaryCreations);
+        given(worldState.getContractNonces()).willReturn(targetContractNonces);
+        given(accountStore.loadAccountOrFailWith(Id.fromGrpcAccount(autoRenewAccount), INVALID_AUTORENEW_ACCOUNT))
+                .willReturn(autoRenewModel);
+        given(autoRenewModel.isSmartContract()).willReturn(false);
+        final var result = TransactionProcessingResult.successful(
+                null,
+                1234L,
+                0L,
+                124L,
+                Bytes.EMPTY,
+                contractAccount.getId().asEvmAddress(),
+                Map.of(),
+                new ArrayList<>());
+        given(txnCtx.consensusTime()).willReturn(consensusTime);
+
+        final var newEvmAddress = contractAccount.getId().asEvmAddress();
+        given(worldState.newContractAddress(senderAccount.getId().asEvmAddress()))
+                .willReturn(newEvmAddress);
+        given(evmTxProcessor.execute(
+                        senderAccount,
+                        contractAccount.getId().asEvmAddress(),
+                        gas,
+                        balance,
+                        Bytes.fromHexString(new String(bytecode)),
+                        txnCtx.consensusTime()))
+                .willReturn(result);
+
+        // when:
+        subject.doStateTransition();
+
+        // then:
+        verify(sigImpactHistorian).markEntityChanged(contractAccount.getId().num());
+        verify(sigImpactHistorian).markEntityChanged(secondaryCreations.get(0).getContractNum());
+        verify(worldState).newContractAddress(senderAccount.getId().asEvmAddress());
+        verify(worldState).setHapiSenderCustomizer(captor.capture());
+        verify(worldState).getCreatedContractIds();
+        verify(worldState).getContractNonces();
+        verify(recordServices).externalizeSuccessfulEvmCreate(result, newEvmAddress.toArrayUnsafe());
+        verify(worldState, never()).reclaimContractId();
```

Comment:
What is this test testing?  If I'm understanding mockito correctly, this is just checking that the call was made - not what the _values_ are, right?